oioioi/forum/views.py

```diff
@@ -52,9 +53,21 @@
 @enforce_condition(contest_exists & can_enter_contest)
 @enforce_condition(forum_exists_and_visible & is_proper_forum)
 def forum_view(request):
-    category_set = request.contest.forum.category_set.prefetch_related(
-        'thread_set', 'thread_set__post_set'
-    ).all()
+    categories = []
+    for category in request.contest.forum.category_set.all():
+        threads = category.thread_set.annotate(
```

Comment:
Here we are creating a database query for every category. How about querying for number of posts in each category in a single query? I could write an SQL for that, so should be possible to do that in django too.